lib/internal/repl.js

```diff
@@ -3,51 +3,35 @@
 const {
   Number,
   NumberIsNaN,
-  ObjectCreate,
 } = primordials;
 
 const REPL = require('repl');
 const { kStandaloneREPL } = require('internal/repl/utils');
 
-module.exports = ObjectCreate(REPL);
-module.exports.createInternalRepl = createRepl;
-
-function createRepl(env, opts, cb) {
-  if (typeof opts === 'function') {
-    cb = opts;
-    opts = null;
-  }
+function createInternalRepl(opts = {}, callback = () => {}) {
   opts = {
     [kStandaloneREPL]: true,
-    ignoreUndefined: false,
     useGlobal: true,
     breakEvalOnSigint: true,
-    ...opts
+    ...opts,
+    terminal: parseInt(process.env.NODE_NO_READLINE) ? false : opts.terminal,
```

Comment:
Should the environment variable override the passed in option?

Reply:
There is no behavior change to the former implementation.

Reply:
Yeah, that I know... I'm just wondering if that's really the behavior we want. :-)

Reply:
It won't matter in this case, since it's only the internal REPL and we only pass these options through in tests.

Reply:
That's fine for now then, but a TODO comment here indicating that it may be worth revisiting it later on would be helpful